images/build/debian-iptables/distroless/Dockerfile

```diff
@@ -0,0 +1,63 @@
+# Copyright 2022 The Kubernetes Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+ARG GORUNNERIMAGE
+# Using directly debian base, as we are going to just copy the required binaries
+FROM debian:bullseye as build
+
+COPY stage-binary-and-deps.sh /
+COPY get-binaries-from-package.sh /
+
+ARG STAGE_DIR="/opt/stage"
+RUN apt -y update && \
+    apt -y install bash curl && \
+    mkdir -p "${STAGE_DIR}" && \
+    /get-binaries-from-package.sh conntrack "${STAGE_DIR}" && \
+    /get-binaries-from-package.sh ebtables "${STAGE_DIR}" && \
+    /get-binaries-from-package.sh ipset "${STAGE_DIR}" && \
+    /get-binaries-from-package.sh iptables "${STAGE_DIR}" && \
+    /get-binaries-from-package.sh kmod "${STAGE_DIR}" && \
+    `# dash is the real Debian basic shell` \
+    /stage-binary-and-deps.sh /bin/dash "${STAGE_DIR}" && \
+    `# below binaries are used by iptables-wrapper-installer.sh` \
+    /stage-binary-and-deps.sh /bin/cat "${STAGE_DIR}" && \
+    /stage-binary-and-deps.sh /bin/chmod "${STAGE_DIR}" && \
+    /stage-binary-and-deps.sh /bin/grep "${STAGE_DIR}" && \
+    /stage-binary-and-deps.sh /bin/ln "${STAGE_DIR}" && \
+    /stage-binary-and-deps.sh /bin/rm "${STAGE_DIR}" && \
+    /stage-binary-and-deps.sh /usr/bin/wc "${STAGE_DIR}" && \
+    `# iptables modules needs to be copied` \
+    cp -r /usr/lib/"$(uname -m)"-linux-gnu/xtables/ "${STAGE_DIR}"/usr/lib/"$(uname -m)"-linux-gnu/ && \
```

Comment:
`get-binaries-from-package.sh` seems to operate on an "assume files in the package are unnecessary unless we can prove that they are necessary" principle, which seems backwards to me. It should copy everything except the things that it knows for sure won't be needed.

Like, if the package installs files into `/etc`, it seems like there's a good chance it may need to read them at runtime...

I worry that the current script will result in an image that passes CI, but that fails if you enable some obscure kube-proxy command-line option, because the image is missing some support file that iptables needs only in certain situations.

Reply:
agreed, I will change the script here to stage all files from a package BUT some specific we know can be discarded (like man, etc)